src/lib/jsonrpc.ml

```diff
@@ -226,7 +226,7 @@ let version_id_and_call_of_string s =
   | None    -> raise (Malformed_method_request "Invalid field 'id' in request body")
 
 let call_of_string str =
-  let _, _, call = version_id_and_call_of_string str in
+  let _, _, call = version_id_and_call_of_string_option str in
```

Comment:
This seems like a change that should at least be in a separate commit.

Reply:
Ah, it is, never mind.

Reply:
So what's calling this? Wouldn't it be better to call the more general function directly?

Reply:
In fact, this one is more general since it returns an option type for id (which we discard here). The `version_id_and_call_of_string` function is a restriction of `version_id_and_call_of_string_option` to valid and present ids.

Reply:
If you're speaking about `call_of_string`, it's not called in ocaml-rpc per se, just in tests and examples. But I use it in my program, for instance.

Reply:
Right, I was thinking that you should be calling `version_id_and_call_of_string_option` directly rather than changing the behaviour of `call_of_string`

Reply:
Okay, reverting it then!